### PR TITLE
LPS-145590 Add error message in workflow with empty Label and Node Id repeated

### PR DESCRIPTION
**Pull request by igor-franca**
https://issues.liferay.com/browse/LPS-145590

### CODE REVIEW

**Comment by M-IgorFranca** [comment]
ci:test:sf

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 1]
```diff
@@ -149,12 +149,29 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
 	const publishDefinition = () => {
 		let alertMessage;
 
-		if (!definitionTitle) {
-			alertMessage = Liferay.Language.get('name-workflow-before-publish');
+		if (!definitionTitle || blockingErrors.errorType !== '') {
+			if (!definitionTitle) {
+				alertMessage = Liferay.Language.get(
+					'name-workflow-before-publish'
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:155]: Hi @M-IgorFranca could you please check if we can simplify this code with code below?

`if (!definitionTitle) {
	alertMessage = Liferay.Language.get('name-workflow-before-publish');

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
} else if (blockingErrors.errorType !== '') {
	switch (blockingErrors.errorType) {
		case 'emptyField':
			alertMessage = Liferay.Language.get('enter-a-label');
			break;
		case 'duplicated':
			alertMessage = Liferay.Language.get(
				'a-node-with-that-id-already-exists'
			);
			break;
		default:
			alertMessage = Liferay.Language.get('error');
	}

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
}`

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 2]
```diff
@@ -149,12 +149,29 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
...
+						break;
+					case 'duplicated':
+						alertMessage = Liferay.Language.get(
+							'a-node-with-that-id-already-exists'
+						);
+						break;
+					default:
+						alertMessage = Liferay.Language.get('error');
+				}
+			}
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:172]: Hi @M-IgorFranca could you please check if we can simplify the code above with code below?
if (!definitionTitle) {
	alertMessage = Liferay.Language.get('name-workflow-before-publish');

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
} else if (blockingErrors.errorType !== '') {
	switch (blockingErrors.errorType) {
		case 'emptyField':
			alertMessage = Liferay.Language.get('enter-a-label');
			break;
		case 'duplicated':
			alertMessage = Liferay.Language.get(
				'a-node-with-that-id-already-exists'
			);
			break;
		default:
			alertMessage = Liferay.Language.get('error');
	}

	setAlertMessage(alertMessage);
	setAlertType('danger');

	setShowAlert(true);
}

Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:172]: Hello @seltong , that would work too, I tried implementing it this way so I don't repeat this code snippet.

`setAlertMessage(alertMessage);
setAlertType('danger');

setShowAlert(true);`

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 3]
```diff
@@ -150,11 +150,30 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
...
 
 		if (!definitionTitle) {
-			alertMessage = Liferay.Language.get('name-workflow-before-publish');
-
+			alertMessage = Liferay.Language.get(
+				'name-workflow-before-publish'
+			);
+			setAlertMessage(alertMessage);
+			setAlertType('danger');
+			setShowAlert(true);
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:158]: Hi @M-IgorFranca what do you think about create a method to make this code lines reusable?

Something like:
```
function _setAlert(alertMessage, alertType, showAlert) {
        setAlertMessage(alertMessage);
	setAlertType(alertType);
	setShowAlert(showAlert);
}
```

If it's possible, we can use this method in everywhere we need it.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 4]
```diff
@@ -65,6 +65,13 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
 		translations,
 		version,
 	} = useContext(DefinitionBuilderContext);
+
+	function _setAlert(alertMessage, alertType, showAlert) {
+		setAlertMessage(alertMessage);
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:70]: Hi @M-IgorFranca I believe that you need to check if the `alertMessage` exists before execute `setAlertMessage(alertMessage);`. Could yout check if it's necessary?

**Comment by naraandrade** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 5]
```diff
@@ -65,6 +65,13 @@ export default function UpperToolbar({displayNames, isView, languageIds}) {
 		translations,
 		version,
 	} = useContext(DefinitionBuilderContext);
+
+	function _setAlert(alertMessage, alertType, showAlert) {
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:69]: @M-IgorFranca Great job :) Could you just remove the `_` in the beginning of the function name? This is a standard followed on our back-end files, but not used in our front-end files. Thanks!

**Review by naraandrade — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 5 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:69]: Of course i can, thanks for the feedback.

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
Rebase this branch with the upstream/master

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Pull request closed without merging**